game/src/core/utils/Direction.java

```diff
@@ -0,0 +1,186 @@
+   * current direction. It returns a new direction based on the transformation:
+   *
+   * <ul>
+   *   <li>DOWN: Returns the opposite direction.
+   *   <li>LEFT: Returns the direction after a left turn.
+   *   <li>RIGHT: Returns the direction after a right turn.
+   *   <li>NONE or UP: Returns the current direction unchanged.
+   * </ul>
+   *
+   * @param relative The relative direction to apply.
+   * @return The new direction after applying the relative transformation.
+   */
+  public Direction applyRelative(Direction relative) {
+    return switch (relative) {
+      case DOWN -> this.opposite();
+      case LEFT -> this.turnLeft();
+      case RIGHT -> this.turnRight();
+      default -> this; // UP or NONE
+    };
+  }
+
+  /**
+   * Returns a random cardinal direction.
+   *
+   * @return A random direction from UP, RIGHT, DOWN, LEFT.
+   */
+  public static Direction random() {
+    Direction[] cardinalDirections = {UP, RIGHT, DOWN, LEFT};
+    return cardinalDirections[RANDOM.nextInt(cardinalDirections.length)];
+  }
+
+  /**
+   * Translates the given coordinate by the direction vector of this Direction.
+   *
+   * <p>This method calculates the new coordinate by adding the direction vector (derived from this
+   * Direction) to the provided coordinate.
+   *
+   * @param coordinate The original coordinate to be translated.
+   * @return A new Coordinate object representing the translated position.
+   */
+  public Coordinate translate(Coordinate coordinate) {
+    Vector<Integer> dirVec = directionVector();
+    return coordinate.add(new Coordinate(dirVec.getFirst(), dirVec.getLast()));
+  }
+
+  /**
+   * Translates the given point by the direction vector of this Direction.
+   *
+   * <p>This method calculates the new point by adding the direction vector (derived from this
+   * Direction) to the provided point.
+   *
+   * @param point The original point to be translated.
+   * @return A new Point object representing the translated position.
+   */
+  public Point translate(Point point) {
+    Vector<Integer> dirVec = directionVector();
+    return point.add(new Point(dirVec.getFirst(), dirVec.getLast()));
+  }
+
+  /**
+   * Returns a vector representation of the direction.
+   *
+   * <p>This method converts the direction into a {@link Vector2} where:
+   *
+   * <ul>
+   *   <li>UP is represented as (0, -1)
+   *   <li>DOWN is represented as (0, 1)
+   *   <li>LEFT is represented as (-1, 0)
+   *   <li>RIGHT is represented as (1, 0)
+   *   <li>NONE is represented as (0, 0)
+   * </ul>
+   *
+   * * @return A {@link Vector2} representing the direction.
+   */
+  private Vector<Integer> directionVector() {
+    int x = 0;
+    int y = 0;
+    switch (this) {
+      case UP -> y = -1;
+      case DOWN -> y = 1;
+      case LEFT -> x = -1;
+      case RIGHT -> x = 1;
+      case NONE -> {
+        // No change to x and y
+      }
+    }
+    return new Vector<>(x, y);
+  }
```

Comment:
Ich habe jetzt diese Methode, erstmal nur private.

Aber jetzt muss ich als Kunde folgendes machen:
https://github.com/Dungeon-CampusMinden/Dungeon/blob/26573c674d37a7bb1af809da0aab35f8fe071994/blockly/src/components/TintDirectionComponent.java#L122-L123

Würde dann nicht sinn machen diese Methode public zu haben?

Plus Vector ist noch nicht der richtige Datentyp. Die Frage ist vlt ob man mal ein Vector2 (einmal für Integer und einmal für Float) erstellt in Core? Aktuell wird an manchen Stellen Point oder Coordinate als Vektor missbraucht.

Reply:
```
      Coordinate dirCoord = direction.translate(new Coordinate(0, 0));
      Coordinate targetCoord = origin.add(new Coordinate(dirCoord.x * i, dirCoord.y * i));
```

sollte das nicht einfach sowas sein wie `direction.translate(origin, i)`?

Reply:
das könnte gehen 🙃

mich stört beim lesen: 

(a) ich addiere hier auf eine koordinate einen vektor. eigentlich verschiebe ich einen punkt/eine koordinate mit einen vektor, also vektor.translate(point) oder zur not point.translate(vector). ... das "add" liest sich komisch. aber wenn am ende womöglich eine koordinate auch nur ein vector2<int> ist, dann ergibt das mehr sinn.

(b) sollte die direction nicht bereits ein vektor sein? ich würde gern das multiply direkt auf der direction aufrufen.

d.h. sollte es nicht so aussehen: `Coordinate targetCoord = direction.multiply(i).translate(origin);`? wobei, dann kann man auch gleich wieder sagen `Coordinate targetCoord = direction.translate(origin, i);` 🤣